src/helpers/customCssStyleSheet.js

```diff
@@ -0,0 +1,44 @@
+import { forEach } from "./forEach";
+import { getCamelCase } from "./getCamelCase";
+export class CustomCssStyleSheet {
+
+  _rules = [];
+
+  _rulesMap = {};
+
+  set rules(value) {
+    this._rules = value;
+  }
+
+  get rules() {
+    return this._rules;
+  }
+
+  set rulesMap(value) {
+    this._rulesMap = value;
+  }
+
+  get rulesMap() {
+    return this._rulesMap;
+  }
+
+  addRule(selector, rules, index) {
+    this.rules[index] = { selector, rules };
```

Comment:
if not index is passed. also if a overriding index is passed. what about the item that was already present there?